### PR TITLE
tests for proof set logic

### PR DESCRIPTION
**Pull request by maxtaco**
(and bugfixes that the tests required!!!)

### CODE REVIEW

File: go/teams/proofs.go [hunk 1]
```diff
@@ -80,9 +78,48 @@ func (p *proofSetT) AddNeededHappensBeforeProof(a proofTerm, b proofTerm) *proof
...
 		}
 	}
 	p.proofs[idx] = append(p.proofs[idx], proof{a, b})
 	return p
 }
+
+func (p *proofSetT) AllProofs() []proof {
+	var ret []proof
+	for _, v := range p.proofs {
+		for _, p := range v {
```
Comment (mlsteele) [inline go/teams/proofs.go]: shadowing `p` here is a little confusing

Comment (maxtaco) [inline go/teams/proofs.go]: indeed sorry about that

File: go/teams/proofs_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+func proofEq(x, y proof) bool {
+	return termEq(x.a, y.a) && termEq(x.b, y.b)
+}
+
+func termEq(a, b proofTerm) bool {
+	return a.leafID == b.leafID && a.sigMeta.SigChainLocation.Seqno == b.sigMeta.SigChainLocation.Seqno
+}
+
+func TestAddNeededHappensBeforeProof(t *testing.T) {
```
Comment (mlsteele) [inline go/teams/proofs_test.go]: I did the crossword puzzle and it seems like there are wires crossed. `e` before `f` and `k` before `l` is impossible?

File: go/teams/proofs_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	ps = ps.AddNeededHappensBeforeProof(q, r)
+
+	ret := ps.AllProofs()
+
+	require.Equal(t, len(ret), 5)
+	require.True(t, proofEq(ret[0], proof{a, h}))
+	require.True(t, proofEq(ret[1], proof{i, d}))
+	require.True(t, proofEq(ret[2], proof{k, l}))
+	require.True(t, proofEq(ret[3], proof{q, r}))
+	require.True(t, proofEq(ret[4], proof{o, p}))
```
Comment (mlsteele) [inline go/teams/proofs_test.go]: I agree! :)

**Pull request merged**